backend/src/modules/authentication/core/domain/UserAccount.ts

```diff
@@ -34,9 +41,37 @@ export function setPasswordForUserAccount(
   };
 }
 
-function onPasswordWasSet(state: UserAccount | undefined, event: PasswordWasSet): UserAccount {
+function onPasswordWasSet(state: UserAccount, event: PasswordWasSet): UserAccount {
   return new UserAccount({
-    email: event.email,
+    userId: event.userId,
+    email: state.email,
     password: event.password,
   });
 }
+
+export async function authenticateUser(
+  state: UserAccount | undefined,
+  command: { email: string; password: string },
+  currentTime: Date,
+): Promise<DomainCommandResult<string | undefined>> {
```

Comment:
Teraz chyba nie moze byc undefined, nie? To gdybysmy wprowadzili event zamiast throw.

Reply:
no w sumie racja :-)